drivers/timer/mcux_os_timer.c

```diff
@@ -194,17 +222,23 @@ void sys_clock_set_timeout(int32_t ticks, bool idle)
 		if (pm_state_next_get(0)->state == PM_STATE_STANDBY) {
 			uint64_t timeout;
 
-			/* Check the amount of time left and switch to a counter
-			 * that is active in this power mode.
-			 */
-			timeout = base->MATCH_L;
-			timeout |= (uint64_t)(base->MATCH_H) << 32;
-			timeout = OSTIMER_GrayToDecimal(timeout);
-			timeout -= OSTIMER_GetCurrentTimerValue(base);
-			/* Round up to the next tick boundary */
-			timeout += (CYC_PER_TICK - 1);
-			/* Convert to microseconds and round up to the next value */
-			timeout = (((timeout / CYC_PER_TICK) * CYC_PER_TICK) * CYC_PER_US);
+			if (wait_forever) {
+				timeout = UINT32_MAX;
+			} else if (counter_overflow) {
+				timeout = counter_remaining_ticks;
+			} else {
+				/* Check the amount of time left and switch to a counter
+				 * that is active in this power mode.
+				 */
+				timeout = base->MATCH_L;
+				timeout |= (uint64_t)(base->MATCH_H) << 32;
+				timeout = OSTIMER_GrayToDecimal(timeout);
+				timeout -= OSTIMER_GetCurrentTimerValue(base);
+				/* Round up to the next tick boundary */
+				timeout += (CYC_PER_TICK - 1);
+				/* Convert to microseconds and round up to the next value */
+				timeout = (((timeout / CYC_PER_TICK) * CYC_PER_TICK) * CYC_PER_US);
+			}
 			if (mcux_lpc_ostick_set_counter_timeout(timeout) == 0) {
```

Comment:
i dont understand something here, why do you pass this value as timeout which then gets converted to ticks, but above you are setting it to a value with unit of ticks, it seems like a bug

Reply:
Converting os timer ticks to counter ticks

Reply:
these names are terribly confusing, this code is made harder to understand by every variable being called the same